command/commands/team.py

```diff
@@ -182,25 +227,219 @@ def create_helper(self, param_list):
         ``None``, will add all members of channel in which the
         command was called into the team.
         :param param_list: List of parameters for creating team
+        :param user_id: Slack ID of user who called command
         :return: return error message if team created unsuccessfully
                     otherwise returns success message
         """
         try:
-            msg = "new team: {}, ".format(param_list["team_name"])
+            command_user = self.facade.retrieve(User, user_id)
+            if not check_credentials(command_user, None):
+                return self.permission_error, 200
+
+            msg = f"New team created: {param_list['team_name']}, "
             team_id = self.gh.org_create_team()
-            team = Team(team_id, param_list["team_name"], "")
+            team = Team(team_id, param_list['team_name'], "")
             if param_list["name"] is not None:
-                msg += "name: {}, ".format(param_list["name"])
-                team.display_name = param_list["name"]
+                msg += f"name: {param_list['name']}, "
+                team.display_name = param_list['name']
             if param_list["platform"] is not None:
-                msg += "platform: {}, ".format(param_list["platform"])
-                team.platform = param_list["platform"]
+                msg += f"platform: {param_list['platform']}, "
+                team.platform = param_list['platform']
             if param_list["channel"] is not None:
-                msg += "add channel"
-                # stub: cannot finish until teamMember PR is pushed
-            self.facade.store_team(team)
+                msg += "added channel, "
+                for member_id in self.sc.get_channel_users(
+                        param_list["channel"]):
+                    member = self.facade.retrieve(User, member_id)
+                    self.gh.add_team_member(member.github_username, team_id)
+            else:
+                self.gh.add_team_member(command_user.github_username, team_id)
+            if param_list["lead"] is not None:
+                msg += "added lead"
+                lead_user = self.facade.retrieve(User, param_list["lead"])
+                team.add_team_lead(lead_user.github_id)
+                if not self.gh.has_team_member(lead_user.github_username,
+                                               team_id):
+                    self.gh.add_team_member(lead_user.github_username, team_id)
+            else:
+                team.add_team_lead(command_user.github_id)
+
+            self.facade.store(team)
             return msg, 200
         except GithubAPIException as e:
             logging.error("team creation unsuccessful")
-            return "Team creation unsuccessful with the following error"\
+            return "Team creation unsuccessful with the following error: "\
```

Comment:
```suggestion
            return f"Team creation unsuccessful with the following error: {e.data}", 200
```

Though you might wanna split it up a little bit.